internal/graph/check.go

```diff
@@ -109,6 +109,7 @@ type checkOutcome struct {
 }
 
 type LocalChecker struct {
+	// NOTE it doesn't have the request's contextual tuples.
```

Comment:
The original design behind LocalChecker was not meant to have a LocalChecker constructed per request. A LocalChecker only needs to be constructed once for the lifetime of the server, and so the Checker doesn't have fields specific per request. It something should be enforced per request, then it should belong in the `ResolveCheck` implementation.

Reply:
See https://github.com/openfga/openfga/pull/948#issuecomment-1690422567

Reply:
> LocalChecker was not meant to be constructed per request originally.

I could change this PR so that every request needs a distinct `LocalChecker`. Then, `LocalChecker.ds` would become `contextualTupleReader` and this comment would go away.